lib/AST/ASTMangler.cpp

```diff
@@ -1451,23 +1451,26 @@ void ASTMangler::appendAnyGenericType(const GenericTypeDecl *decl) {
   addSubstitution(key.getPointer());
 }
 
-void ASTMangler::appendFunction(AnyFunctionType *fn, bool isFunctionMangling) {
+void ASTMangler::appendFunction(AnyFunctionType *fn, bool isFunctionMangling,
+                                bool captureLabels) {
   // Append parameter labels right before the signature/type.
-  auto parameters = fn->getParams();
-  auto firstLabel = std::find_if(
-                  parameters.begin(), parameters.end(),
-                  [&](AnyFunctionType::Param param) { return param.hasLabel(); });
-
-  if (firstLabel != parameters.end()) {
-    for (auto param : parameters) {
-      auto label = param.getLabel();
-      if (!label.empty())
-        appendIdentifier(label.str());
-      else
-        appendOperator("_");
+  if (captureLabels) {
+    auto parameters = fn->getParams();
+    auto firstLabel = std::find_if(
+        parameters.begin(), parameters.end(),
+        [&](AnyFunctionType::Param param) { return param.hasLabel(); });
+
+    if (firstLabel != parameters.end()) {
+      for (auto param : parameters) {
+        auto label = param.getLabel();
+        if (!label.empty())
+          appendIdentifier(label.str());
+        else
+          appendOperator("_");
+      }
+    } else if (parameters.size() > 0) {
+      appendOperator("y");
     }
```

Comment:
you could add an assert in the else of if(captureLabels) to confirm that there are actually no labels in the function signature